jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/Query.java

```diff
@@ -1,13 +1,36 @@
 package dev.jlibra.admissioncontrol.query;
 
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 import org.immutables.value.Value;
 
+import types.GetWithProof.RequestItem;
+
 @Value.Immutable
 public interface Query {
 
-    List<GetAccountState> getAccountStateQueries();
+    Optional<List<GetAccountState>> getAccountStateQueries();
+
+    Optional<List<GetAccountTransactionBySequenceNumber>> getAccountTransactionBySequenceNumberQueries();
+
+    default List<RequestItem> toGrpcObject() {
+        List<RequestItem> requestItems = new ArrayList<>();
+
+        if (getAccountStateQueries().isPresent()) {
+            requestItems.addAll(getAccountStateQueries().get().stream().map(GetAccountState::toGrpcObject)
+                    .collect(toList()));
+        }
+
+        if (getAccountTransactionBySequenceNumberQueries().isPresent()) {
+            requestItems.addAll(getAccountTransactionBySequenceNumberQueries().get().stream()
+                    .map(GetAccountTransactionBySequenceNumber::toGrpcObject)
+                    .collect(toList()));
+        }
 
-    List<GetAccountTransactionBySequenceNumber> getAccountTransactionBySequenceNumberQueries();
+        return requestItems;
```

Comment:
Just a suggestion without garbage variable
```
    default List<RequestItem> toGrpcObject() {
        return Stream.concat(
                getAccountStateQueries()
                        .map(Collection::stream)
                        .orElse(Stream.empty())
                        .map(GetAccountState::toGrpcObject),
                getAccountTransactionBySequenceNumberQueries().map(Collection::stream)
                        .orElse(Stream.empty())
                        .map(GetAccountTransactionBySequenceNumber::toGrpcObject)
        ).collect(toList());
    }
```

Reply:
Thanks, I did not know it could be done like that :+1:

Reply:
That's nice. And even more readable.